whois-query/src/main/java/net/ripe/db/whois/query/rpki/RoutinatorDataProvider.java

```diff
@@ -0,0 +1,46 @@
+package net.ripe.db.whois.query.rpki;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.jakarta.rs.json.JacksonJsonProvider;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.MediaType;
+import net.ripe.db.whois.common.profiles.WhoisProfile;
+import org.glassfish.jersey.client.ClientProperties;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+@Profile({WhoisProfile.DEPLOYED})
+public class RoutinatorDataProvider implements RpkiDataProvider{
+
+    private static final int CLIENT_CONNECT_TIMEOUT = 20_000;
+    private static final int CLIENT_READ_TIMEOUT = 120_000;
+
+    private final Client client;
+    private final String rpkiBaseUrl;
+
+    public RoutinatorDataProvider(@Value("${rpki.base.url}") final String rpkiBaseUrl) {
```

Comment:
This client becomes *mandatory* if you have to set `rpki.base.url`. Instead make it optional and fail if the client is used.

Reply:
I made it optional and return a null with an error log.
Upper class is throwing an error if it gets a null from that call.
I marked the method as nullable as well

Reply:
Thanks